src/v3/src/util/settingsUtils.ts

```diff
@@ -246,3 +248,33 @@ export const isOauth2Enabled = (widgetProps: WidgetProps): boolean => {
   }
   return typeof clientId !== 'undefined' && authScheme?.toLowerCase() === 'oauth2';
 };
+
+export const getPageTitle = (
+  widgetProps: WidgetProps,
+  formTitle: string,
+  idxTransaction?: IdxTransaction,
+): string | null => {
+  const { brandName, features: { setPageTitle } = {} } = widgetProps;
+  const eventContext: EventContext = typeof idxTransaction === 'undefined'
+    ? { controller: null }
+    : getEventContext(idxTransaction);
+
+  switch (typeof setPageTitle) {
+    // When setPageTitle option is 'undefined', default will be to set title based on page header
+    // When setPageTitle option is 'true', set title based on page header
+    case 'boolean':
+    case 'undefined':
+      if (setPageTitle === false) {
+        // Indicates setPageTitle config option was purposefully disabled
+        return null;
+      }
+      return brandName ? `${brandName} | ${formTitle}` : formTitle;
+    case 'string':
+      return setPageTitle;
+    case 'function':
+      return setPageTitle(eventContext, { formTitle, brandName });
+    default:
+      // Indicates an invalid/unexpected value was passed into the config option
+      return null;
```

Comment:
Maybe? Or would using `console.error` suffice? throwing here seems like it would be intrusive since this isnt really a functional component of the app

Reply:
Worth bringing up in standup, maybe we can wait for more of a holistic config validation and just error or warn here.